docs/api.md

```diff
@@ -0,0 +1,163 @@
+# API Request Structure
+
+## `/bid`
+
+Returns encoded data to be signed, a generated auctionId,
+and a generated nftId determined by the NFT contract address and tokenId
+
+```
+POST <base-uri>/api/bid
+
+Headers:
+  Content-Type: application/json
+
+Body:
+{
+  "bidAmount": <string>,
+  "contractAddress": <string>,
+  "tokenId": <string>,
+  "minimumBid": <string>,
+  "startBlock": <string>,
+  "expireBlock": <string>
+}
+
+Response:
+{
+  "payload": <string>,
+  "auctionId": <number>,
+  "nftId": <string>
+}
+```
+
+## `/bids/list`
+
+Endpoint to return auctions associated with each given nftId
+
+```
+POST <base-uri>/api/bids/list
+
+Headers:
+  Content-Type: application/json
+
+Body:
+{
+  "nftIds": [
+     <string>,
+     <string>,
+     ...
+   ]
+}
+
+Response:
+{
+  "0x123...": [
+     "account": <string>,
+     "signedMessage": <string>,
+     "auctionId": <string>,
+     "bidAmount": <string>,
+     "minimumBid": <string>,
+     "startBlock": <string>,
+     "expireblock": <string>,
+     "date": <number>,
+     "tokenId": <string>,
+     "contractAddress": <string>
+   ],
+   ...
+}
+```
+
+## `/bids/accounts/:account`
+
+Endpoint to return all bids by a single given account
+
+```
+GET <base-uri>/api/bids/accounts/:account
+
+Headers:
+  Content-Type: application/json
+
+Body: N/A
+
+Response:
+[
+  {
+    "account": <string>,
+    "signedMessage": <string>,
+    "auctionId": <string>,
+    "bidAmount": <string>,
+    "minimumBid": <string>,
+    "startBlock": <string>,
+    "expireblock": <string>,
+    "date": <number>,
+    "tokenId": <string>,
+    "contractAddress": <string>
+  },
+  ...
+]
+```
+
+## `/bids`
+
+Creates a new bid for an auction once signed by user
+
+```
+POST <base-uri>/api/bids
+
+Headers:
+  Content-Type: application/json
+
+Body: 
+{
+  "account": <string>,
+  "auctionId": <string>,
+  "bidAmount": <string>,
+  "contractAddress": <string>
+  "expireblock": <string>,
+  "minimumBid": <string>,
+  "signedMessage": <string>,
+  "startBlock": <string>,
+  "tokenId": <string>,
+}
+Response:
```

Comment:
Nit: For the future, including examples of failure responses would be helpful.

Reply:
Wrote it down as a todo 😄